src/app/OrderBook.tsx

```diff
@@ -2,29 +2,130 @@ import { useContext } from "react";
 import { OrderbookLine } from "alphadex-sdk-js";
 import { AdexStateContext } from "./contexts";
 
-export function OrderBook({ orders }: { orders: OrderbookLine[] }) {
-  //returns simple orderbook of buys/sells
+interface OrderBookRowProps {
+  barColor: string;
+  order: OrderbookLine;
+}
+
+function OrderBookRow(props: OrderBookRowProps) {
+  const { barColor, order } = props;
+
+  return (
+    <tr className="border-none">
+      <td>{order.noOrders}</td>
+      <td
+        className={
+          (barColor === "green" ? "text-green-700" : "text-red-700") +
+          " text-end"
+        }
+      >
+        {order.price}
+      </td>
+      <td className="text-end">{order.quantityRemaining}</td>
+      <td className="text-end">{order.valueRemaining}</td>
+    </tr>
+  );
+}
+
+interface MiddleRowsProps {
+  // TODO: how to determin if there were no trades to make this field nulllable?
+  // adexState.currentPairInfo.lastPrice returns -1 if there were no trades
+  lastPrice: number;
+  bestSell: number | null;
+  bestBuy: number | null;
+}
+
+function MiddleRows(props: MiddleRowsProps) {
+  const { lastPrice, bestSell, bestBuy } = props;
+
+  if (bestBuy && bestSell && bestBuy !== 0) {
+    const spread = bestSell - bestBuy;
+    const spreadPercent = (spread / bestBuy) * 100;
```

Comment:
I'll include using Fred's utils functions for rounding.

"Do they update automatically when orders get bought?" - they all depend on the adex state, so I guess yes. I'll try to test this.

About the `spreadPercent` I don't really know... maybe 🤔 I just took Fred's word for it (cc @fliebenberg ), which was:
> It is indeed the difference between the best buy and sell prices. You already have this info in the orderbook, so the frontend just needs to calculate the difference between the two values. It can then also calculate the % spread, which is just the spread / best buy price.

from https://www.figma.com/file/P7pfzKwJ4G6ClapXfl61D3?node-id=18:3324&mode=design#505524187

Reply:
Looks the same, but this one would be `spread / bestSell` (not `spread / bestBuy` as in the code), gpt says it can be either.

https://www.wallstreetprep.com/knowledge/bid-ask-spread/
![Screenshot 2023-07-27 at 16 41 19](https://github.com/DeXter-on-Radix/website/assets/27793901/7f442bfd-3568-4d90-88e3-4cd60ecf95e9)

Reply:
I like `(2* spread) / (bestBuy+bestSell)` better for the `percentSpread`, @fliebenberg  what do you think?

Reply:
Using spread/ask price (best sell) will make your spread % lower, which is probably why many exchanges prefer that. I prefer the more conservative option which is spread/Bid (best buy). The (2 * spread)/bestBuy + bestSell) is a good compromise between the 2. I'd say lets go with that for now. I dont think it is a massively important number (it is more of a general indicator of pair efficiency) and we can always reconsider once we have users using the app and they give us feedback on this.